src/XR/features/WebXRHandTracking.ts

```diff
@@ -1,893 +1,866 @@
-            Tags.AddTagsTo(touchMesh, "touchEnabled");
-        }
-
-        const handedness = xrController.inputSource.handedness;
-        const webxrHand = new WebXRHand(
-            xrController,
-            this._handResources.jointMeshes[handedness],
-            this._handResources.handMeshes && this._handResources.handMeshes[handedness],
-            this._handResources.rigMappings && this._handResources.rigMappings[handedness],
-            touchMesh,
-            this.options.handMeshes?.meshesUseLeftHandedCoordinates,
-            this.options.jointMeshes?.invisible,
-            this.options.jointMeshes?.scaleFactor
-        );
-
-        this._attachedHands[xrController.uniqueId] = webxrHand;
-        this._trackingHands[handedness] = webxrHand;
-
-        this.onHandAddedObservable.notifyObservers(webxrHand);
-    }
-
-    private _detachHandById(controllerId: string) {
-        const hand = this.getHandByControllerId(controllerId);
-        if (hand) {
-            const handedness = hand.xrController.inputSource.handedness == "left" ? "left" : "right";
-            if (this._trackingHands[handedness]?.xrController.uniqueId === controllerId) {
-                this._trackingHands[handedness] = null;
-            }
-            this.onHandRemovedObservable.notifyObservers(hand);
-            hand.dispose();
-            delete this._attachedHands[controllerId];
-        }
-    }
-
-    private _detachHand = (xrController: WebXRInputSource) => {
-        this._detachHandById(xrController.uniqueId);
-    }
-
-    /**
-     * Detach this feature.
-     * Will usually be called by the features manager.
-     *
-     * @returns true if successful.
-     */
-    public detach(): boolean {
-        if (!super.detach()) {
-            return false;
-        }
-
-        Object.keys(this._attachedHands).forEach((uniqueId) => this._detachHandById(uniqueId));
-
-        return true;
-    }
-
-    /**
-     * Dispose this feature and all of the resources attached.
-     */
-    public dispose(): void {
-        super.dispose();
-        this.onHandAddedObservable.clear();
-        this.onHandRemovedObservable.clear();
-
-        if (this._handResources.handMeshes && !this.options.handMeshes?.customMeshes) {
-            this._handResources.handMeshes.left.dispose();
-            this._handResources.handMeshes.right.dispose();
-        }
-
-        if (this._handResources.jointMeshes) {
-            this._handResources.jointMeshes.left.forEach((trackedMesh) => trackedMesh.dispose());
-            this._handResources.jointMeshes.right.forEach((trackedMesh) => trackedMesh.dispose());
-        }
-    }
-}
-
-//register the plugin
-WebXRFeaturesManager.AddWebXRFeature(
-    WebXRHandTracking.Name,
-    (xrSessionManager, options) => {
-        return () => new WebXRHandTracking(xrSessionManager, options);
-    },
-    WebXRHandTracking.Version,
-    false
-);
+import { WebXRAbstractFeature } from "./WebXRAbstractFeature";
```

Comment:
nit: Was this just some git weirdness? Seems like if you git some resetting and rebasing you could get rid of this diff (and maybe keep the blame record in-tact)

Reply:
Wow, didn't notice that... In Codeflow it correctly shows only a few small changes.

Reply:
For some reason Sublime was saving only this file with Unix-style line endings...